CommonCompilerConfig.cmake

```diff
@@ -102,6 +104,22 @@ endif()
 message(STATUS "CMAKE BITNESS=${BITNESS}")
 
 
+# Check memory limits (in megabytes)
+if(CMAKE_MAJOR_VERSION GREATER 2)
+  cmake_host_system_information(RESULT total_physical_memory QUERY TOTAL_PHYSICAL_MEMORY)
+  cmake_host_system_information(RESULT total_virtual_memory QUERY TOTAL_VIRTUAL_MEMORY)
+  math(EXPR total_memory "${total_physical_memory}+${total_virtual_memory}")
```

Comment:
Available can be used, rather than total. Depending on the Linux distro used with the RPi, you either get no swap space/file (ubuntuMATE) or a small swap file (~100MB for Raspbian Jessie). Changing to use available will require the warning message to be rephrased, and as breznak suggests even changing it from a WARNING to an ERROR.